Nitpretty: make the tool more customizable

Morriar opened the pull request:
Added some options to control behavior.

Interesting features:

* option --break-strings to enable literal string breaks
* option --inline-do to enable do inlining
* option --skip-empty to enable empty line skipping

privat commented on tests/sav/nitpretty_args1.res:
```diff
@@ -12,4 +12,4 @@
 # limitations under the License.
 
-# An empty module
+# An empty module
```
ajouter un \n sur la derniere ligne pour faire plaisir a git?

Morriar commented:
Reroll avec tes commentaires:

1. j'ai retiré le commit qui enlevait le dernier `\n`
2. le dernier commit s'occupe de rajouter un `\n` uniquement si `--skip-empty` est activé

privat commented:
+1

Morriar commented:
Ping @xymus, @R4PaSs, @BlackMinou go mettre un +1.

R4PaSs commented on src/nitpretty.nit:
```diff
@@ -32,6 +32,16 @@ redef class ToolContext
+	var opt_break_str = new OptionBool("Break too long string literals", "--break-strings")
+
+	# Force `do` on same line than method signature.
+	var opt_inline_do = new OptionBool("Force do keyword on the same line than the method signature",
```
:s/same line than/same line as

R4PaSs commented on src/nitpretty.nit:
```diff
@@ -32,6 +32,16 @@ redef class ToolContext
+	# Break too long string literals.
+	var opt_break_str = new OptionBool("Break too long string literals", "--break-strings")
+
+	# Force `do` on same line than method signature.
```
:s/line than/line as

R4PaSs commented on src/pretty.nit:
```diff
@@ -223,6 +223,14 @@ class PrettyPrinterVisitor
+	end
+
+	# Force the line to end even is not skip_empty.
+	fun forcen do
```
Forcen ? Strange name for a method, I don't understand its purpose either, event with the comment, care to explain ?

R4PaSs commented on src/pretty.nit:
```diff
@@ -243,6 +251,15 @@ class PrettyPrinterVisitor
+	# Do we break string literals that are too long?
+	var break_strings = false is public writable
+
+	# Do we force `do` on the same line than method signature?
```
:s/than method/as the method

xymus commented:
I don't know exactly what the option `--skip-empty` does, but besides that +1.

R4PaSs commented:
+1 aside from my comments

Morriar commented:
Reroll:

* removed useless empty line at pretty.nit:838
* corrected typos as denoted by @R4PaSs 
* changed comment for `forcen`
* enhanced comment for `--skip-empty`
